phoenix-core/src/main/java/org/apache/phoenix/compile/ExpressionCompiler.java

```diff
@@ -1339,4 +1363,188 @@ public Expression visit(SubqueryParseNode node) throws SQLException {
+    		throw TypeMismatchException.newException(children.get(0).getDataType(), children.get(1).getDataType());
+       	}
+    	if(!(PJson.INSTANCE.isComparableTo(children.get(1).getDataType()))){
+      		 throw TypeMismatchException.newException(children.get(0).getDataType(), children.get(1).getDataType());
+       	}
+       	LiteralExpression rhs=(LiteralExpression)children.get(1);
+        Expression expression = new JsonSubsetExpression(children);
+        return wrapGroupByExpression(expression);
+    }
+    
+    @Override
+    public Expression visitLeave(JsonSupersetParseNode node, List<Expression> children) throws SQLException {
+    	if(!(PJson.INSTANCE.isComparableTo(children.get(0).getDataType()))){
+    		throw TypeMismatchException.newException(children.get(0).getDataType(), children.get(1).getDataType());
+       	}
+    	if(!(PVarchar.INSTANCE.isComparableTo(children.get(1).getDataType()))){
+   		 throw TypeMismatchException.newException(children.get(0).getDataType(), children.get(1).getDataType());
+    	}
+    	LiteralExpression rhs=(LiteralExpression)children.get(1);
+        Expression expression = new JsonSupersetExpression(children);
+        return wrapGroupByExpression(expression);
+    }
+    
+    @Override
+    public Expression visitLeave(JsonMultiKeySearchOrParseNode node, List<Expression> children) throws SQLException {
+    	if(!(PJson.INSTANCE.isComparableTo(children.get(0).getDataType()))){
+    		throw TypeMismatchException.newException(children.get(0).getDataType(), children.get(1).getDataType());
+       	}
+    	if(children.get(1).getDataType()!=PVarcharArray.INSTANCE){
+    		 throw TypeMismatchException.newException(children.get(0).getDataType(), children.get(1).getDataType());
+    	}
+        Expression expression = new JsonMultiKeySearchOrExpression(children);
+        return wrapGroupByExpression(expression);
+    }
+    
+    @Override
+    public Expression visitLeave(JsonMultiKeySeatchAndParseNode node, List<Expression> children) throws SQLException {
+    	if(!(PJson.INSTANCE.isComparableTo(children.get(0).getDataType()))){
+    		throw TypeMismatchException.newException(children.get(0).getDataType(), children.get(1).getDataType());
+       	}
+    	if(children.get(1).getDataType()!=PVarcharArray.INSTANCE){
+   		 throw TypeMismatchException.newException(children.get(0).getDataType(), children.get(1).getDataType());
+    	}
+        Expression expression = new JsonMultiKeySearchAndExpression(children);
+        return wrapGroupByExpression(expression);
+    }
+    
+    @Override
+    public Expression visitLeave(JsonPathAsTextParseNode node, List<Expression> children) throws SQLException {
+    	if(!(PJson.INSTANCE.isComparableTo(children.get(0).getDataType()))){
+    		throw TypeMismatchException.newException(children.get(0).getDataType(), children.get(1).getDataType());
+       	}
+    	if(!(PVarchar.INSTANCE.isComparableTo(children.get(1).getDataType()))){
```

Comment:
Because the grammar in postgreSQL of #> and #>> will be
````
 json_column #> '{key,index....}'
 json_column #>> '{key,index...}'
````
is different with the type of varchar array , so we need to add some decode function to decode these format